spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -2181,20 +2181,28 @@ public void testTableWithInt96Timestamp() throws IOException {
             stagingLocation);
 
         // validate we get the expected results back
-        List<Row> expected = spark.table("parquet_table").select("tmp_col").collectAsList();
-        List<Row> actual =
-            spark
-                .read()
-                .format("iceberg")
-                .load(loadLocation(tableIdentifier))
-                .select("tmp_col")
-                .collectAsList();
-        assertThat(actual).as("Rows must match").containsExactlyInAnyOrderElementsOf(expected);
+        testWithFilter("tmp_col < to_timestamp('2000-01-31 08:30:00')", tableIdentifier);
```

Comment:
these tests are all passing without the fix. Can you please add a test that reproduces the issue from https://github.com/apache/iceberg/issues/8990?

Reply:
Filter expression was no longer applied after last rebase. It's fixed now. Please try again.

Reply:
Fixed now.

Reply:
I think it would be good to update the same test across all Spark versions, because the code fixes it already across versions

Reply:
Done